Make webinar highlights actually contain webinar highlights.

douglaschan32167 opened the pull request:
## Proposed changes
Title
Closing #341 
## Types of changes

- [x] I've labeled this PR with appropriate labels (`comp/*` component, bug/feature, documentation and other less important ones).

## Checklist
_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging your code._

- [x] I've linked the proper Github Issue(s) to my PR
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests (unit, integration...etc.) that prove my fix is effective or that my change works
- [x] I have added or updated necessary documentation (if appropriate). You can search, for instance using the [`silver searcher`](https://github.com/ggreer/the_silver_searcher) `ag <STRING>` to find all usages in the repository.
- [x] I have notified the team of any breaking or important changes

vickyadebona commented on frontend/src/pages/Blogs/Blogpost.jsx:
```diff
@@ -343,11 +344,12 @@ class Blogpost extends Component {
     // an image or video for the header
     if (headerMedia.includes('youtube')) {
-      const videoId = this.parseVideoLink(headerMedia);
+      const videoId = ParseYoutubeID(headerMedia);
```
I think we can delete the `parseVideoLink()` function if we're using a new function for parsing YouTube links.

douglaschan32167 replied:
done

vickyadebona commented on frontend/src/styles/globalStyle.js:
```diff
@@ -112,9 +112,9 @@ const GlobalStyle = createGlobalStyle`
-  img:not([alt]) {
-    filter: blur(5px);
-  }
+  // img:not([alt]) {
+  //   filter: blur(5px);
+  // }
 
```
If this code is not going to be used anytime soon, then we can delete it.

douglaschan32167 replied:
removed.

dbtung commented on frontend/src/utils/youtubeIdParser.js:
```diff
@@ -0,0 +1,12 @@
+        ID = url;
+    }
+    return ID;
+}
```
nit: End file with a newline.

dbtung commented on frontend/src/components/WebinarHighlight/WebinarHighlight.jsx:
```diff
@@ -3,28 +3,31 @@ import { useTranslation } from 'react-i18next';
 import { Link } from 'react-router-dom';
 import { Embed } from 'semantic-ui-react';
 import styled from 'styled-components';
+import {ParseYoutubeID} from "utils/youtubeIdParser";
+import {logContentClick} from "utils/google_tag_manager_helpers";
```
nit: Should have space after `{` and before `}`

dbtung commented on frontend/src/components/WebinarHighlight/WebinarHighlight.jsx:
```diff
@@ -3,28 +3,31 @@ import { useTranslation } from 'react-i18next';
-const WebinarHighlight = ({ title, id, blog }) => {
+const WebinarHighlight = ({ webinar, index }) => {
   const { t } = useTranslation(['webinar']);
+    const youtubeId = ParseYoutubeID(webinar.body_content);
```
Indentation looks off for this line.

dbtung commented on frontend/src/pages/Blogs/Blogpost.jsx:
```diff
@@ -14,6 +14,7 @@ import theme from '../../styles/theme';
 
 import arrowLeft from '../../images/arrow-left-purple.png';
 import { logContentView } from 'utils/google_tag_manager_helpers';
+import {ParseYoutubeID} from "utils/youtubeIdParser";
```
nit: Add spaces

dbtung commented on frontend/src/pages/Home/Home.jsx:
```diff
@@ -36,14 +37,30 @@ const Home = () => {
     getFeaturedBlogPosts();
   }, []);
 
+    useEffect(() => {
+        const getRecentWebinars = async () => {
+            await BlogPosts.getWebinars()
+                .then(results => {
+                    setRecentWebinars(results.results);
+                })
+                .catch(error => {
+                    const fallbackWebinars = blogPostFallbackData.filter(
+                        blogpostContentJson => blogpostContentJson.blogpost.type === "webinar"
+                    );
+                    setRecentWebinars(fallbackWebinars);
+                });
+        };
+        getRecentWebinars();
+    }, []);
```
Indentation looks off here.

douglaschan32167 replied:
Tried to fix it to 2 spaces. Hopefully the linter will do its thing too when I commit

dbtung replied:
Hmmm, the linter should only run on files that are staged for commit - so if it didn't catch it before you committed and pushed to your branch, it might not =(

Worst case we can just run `prettier` on the whole directory later.

douglaschan32167 replied:
I think I made enough changes in the commit that it ran.

douglaschan32167 commented:
I addressed the comments and also made the view all for the webinar highlights section link to the blog list page filtered for webinars. It's not elegant but it works

douglaschan32167 commented:
@vickyadebona @dbtung ready for review. Thanks for doing this!

dbtung commented on frontend/src/utils/youtubeIdParser.js:
```diff
@@ -0,0 +1,12 @@
+// adapted from https://gist.github.com/takien/4077195
+export const ParseYoutubeID = url => {
```
Sorry, this is a nit, but I think this should be camel-cased - so lowercase "P".

douglaschan32167 replied:
done.
